core/src/main/scala-3.x/src/main/scala/cats/syntax/MonadOps.scala

```diff
@@ -0,0 +1,12 @@
+package cats.syntax
+
+import cats.{Monad, Alternative}
+
+final class MonadOps[F[_], A](private val fa: F[A]) extends AnyVal {
```

Comment:
Had to split this out due to https://github.com/lampepfl/dotty/issues/9480

Reply:
I actually don't know how dotty compiles this down. Is `explicitly` still a thing? Do you have a suggestion on how to fix this given the limitation above?

Reply:
I would summon @smarter actually if I hit this. Probably worth looking at the bytecode quickly to see what the parameter order is. Can we simply put the `using` block second?

Reply:
It's source-compatible (to explicitly pass parameters to a using block, you have to write `using` at use-site in front of the block too). I suggested putting the using block first because doing the implicit search requires instantiating type variables which in turns means that in `increment.whileM_(inspect(i => i > 4)).run(3)` we can figure out that `i` needs to be `Int`, see https://github.com/lampepfl/dotty/issues/9480#issuecomment-668545232 for the full explanation.

Reply:
Ahhh, I see. Okay I'm in favor, though I think it might be best if we also do this for all of the other Ops classes.

Reply:
That would increase the maintenance burden though. But if you did want to rewrite all the ops classes for dotty, then you might as well use extension syntax which would look more natural:
```scala
extension [F[_], A](fa: F[A])(using M: Monad[F]) {
  def whileM_(p: F[Boolean]): F[Unit] = ???
  def untilM(...)
}
```

Reply:
I'm just concerned that we're going to end up hitting something like this again in the future, but once we release on Dotty, we can't futz with the binary compatibility any further.

Reply:
Well, you can break BC every time we break BC which is every six weeks until 3.0 is out probably, but afterwards it gets tricky yeah :) (though I guess you can always play trick with something like a package-protected overload to preserve BC?)

Reply:
Are we promising binary compatibility for dotty artifacts? Should we? (We certainly don't test it yet)

Reply:
I don't think we're promising BC until Scala 3 goes final. Once it does though, we 100% should be promising it.